src/OpenTelemetry/BatchExportProcessor.cs

```diff
@@ -57,12 +56,9 @@ protected BatchExportProcessor(
         this.ScheduledDelayMilliseconds = scheduledDelayMilliseconds;
         this.ExporterTimeoutMilliseconds = exporterTimeoutMilliseconds;
         this.MaxExportBatchSize = maxExportBatchSize;
-        this.exporterThread = new Thread(this.ExporterProc)
-        {
-            IsBackground = true,
-            Name = $"OpenTelemetry-{nameof(BatchExportProcessor<T>)}-{exporter.GetType().Name}",
-        };
-        this.exporterThread.Start();
+        this.exportTask = Task.CompletedTask;
+        this.exporterTaskCancellation = new CancellationTokenSource();
+        this.exporterTask = Task.Run(this.ExporterProc);
```

Comment:
I think we shouldn't change the default behavior here. If there are scenarios where a thread pool must be leveraged, then it can be a optional, opt-in behavior - either a modification for the BatchExportProcessor (UseDedicatedThread=false) or a new `BatchExportProcessorThreadPool`.

Having said that, I'll not have bandwidth to do a detailed review at the moment, but I believe the first step is to get some guidance from the maintainers about which direction should be taken:
1. Provide a opt-in feature flag to BatchExportProcessor so it can do dedicated thread or threadpool.
2. Make a new BatchExportProcessor to do the thread pool one - either in this repo OR in the contrib repo.
3. Something else.

Reply:
That's a good idea. I'll do some experimentations.